### PR TITLE
Check for null

### PR DESCRIPTION
**Pull request by ovalseven8**
Hey, no idea if you like such checks. :smile: 

Just skipped through the code and added it for the important public APIs. Just close if you don't think it's useful. ;-)

### CODE REVIEW

**Comment by lgrahl** [comment]
Thanks for the effort. From my perspective, the arguments are clearly annotated with `@NonNull`. I consider `@NonNull` a contract that needs to be fulfilled by the application/library using the API. And thus it should not be required to check for `null`.

Conversely, you would not wrap all method calls into try/catch even though they could theoretically raise an undocumented runtime exception.

Your thoughts, @dbrgn?

File: src/main/java/org/saltyrtc/client/cookie/CookiePair.java [hunk 1]
```diff
@@ -54,13 +56,9 @@ public Cookie getOurs() {
...
     }
 
-    public boolean hasTheirs() {
-        return this.theirs != null;
-    }
-
     @Nullable
-    public Cookie getTheirs() {
-        return this.theirs;
+    public Optional<Cookie> getTheirs() {
```
Comment (lgrahl) [inline src/main/java/org/saltyrtc/client/cookie/CookiePair.java]: IIRC we cannot use `Optional` because older Android versions need to be supported.

Comment (dbrgn) [reply src/main/java/org/saltyrtc/client/cookie/CookiePair.java]: Yep, very unfortunate but true.

**Comment by dbrgn** [comment]
I agree with @lgrahl. The @Nullable and @NonNull annotations are a contract, if that is violated a NPE may be raised.

More such annotations are welcome, as long as they are correct :slightly_smiling_face:

File: src/main/java/org/saltyrtc/client/cookie/CookiePair.java (hunk 1 above)
Comment (ovalseven8) [reply src/main/java/org/saltyrtc/client/cookie/CookiePair.java]: What about TaskHelper?
https://github.com/saltyrtc/saltyrtc-client-java/blob/87fd191daae0f2ff727de497f218609d90facda4/src/main/java/org/saltyrtc/client/helpers/TaskHelper.java#L41

Comment (lgrahl) [reply src/main/java/org/saltyrtc/client/cookie/CookiePair.java]: Must have been an oversight. Unless this can now be desugared for API level < 24.

**Pull request closed without merging**